core/src/validator.rs

```diff
@@ -175,7 +180,37 @@ impl Validator {
             config.storage_slots_per_turn,
             bank.slots_per_segment(),
         );
-
+        let bank_forks = bank_forks.clone();
+        let counter = Arc::new(AtomicUsize::new(0));
```

Comment:
These two lines can move after `if config. enable_ctrl_c_handler {` to keep their scope as limited as possible